onnxruntime/core/framework/allocatormgr.cc

```diff
@@ -12,21 +12,37 @@
 namespace onnxruntime {
 using namespace common;
 
-AllocatorPtr CreateAllocator(const DeviceAllocatorRegistrationInfo& info,
-                             OrtDevice::DeviceId device_id, bool use_arena) {
-  auto device_allocator = std::unique_ptr<IDeviceAllocator>(info.factory(device_id));
+AllocatorPtr CreateAllocator(const AllocatorCreationInfo& info) {
+  auto device_allocator = std::unique_ptr<IDeviceAllocator>(info.device_alloc_factory(info.device_id));
+
+  if (info.use_arena) {
+    size_t max_mem = info.arena_cfg.max_mem == -1 ? BFCArena::DEFAULT_MAX_MEM : info.arena_cfg.max_mem;
+    int initial_chunk_size_bytes = info.arena_cfg.initial_chunk_size_bytes == -1
+                                       ? BFCArena::DEFAULT_INITIAL_CHUNK_SIZE_BYTES
+                                       : info.arena_cfg.initial_chunk_size_bytes;
+    int max_dead_bytes_per_chunk = info.arena_cfg.max_dead_bytes_per_chunk == -1
+                                       ? BFCArena::DEFAULT_MAX_DEAD_BYTES_PER_CHUNK
+                                       : info.arena_cfg.max_dead_bytes_per_chunk;
+    ArenaExtendStrategy arena_extend_str;
+    switch (info.arena_cfg.arena_extend_strategy) {
+      case 1:
+        arena_extend_str = ArenaExtendStrategy::kSameAsRequested;
```

Comment:
nit: slightly opaque to have '1' as the case vs `case static_cast<int>(ArenaExtendStrategy::kSameAsRequested):`